Feature/15356 invite link

aerych opened the pull request:
Refs #15356 

This PR wires up the UI, analytics, and associated behaviors for the Invite Links feature.

Changes include:
- An invite links section that appears only for P2s
- An option to generate links if there are none. 
- Options to share, configure, or disable existing links.
- Errors should be displayed in a "toast" notice.
- Tracks events that should match [Android](https://github.com/wordpress-mobile/WordPress-Android/pull/14045)

Note: 
- The share cell is missing an share icon. I didn't want to hold up the PR for this minor detail.  To support the icon, the cell is going to need to be rebuilt and I'd rather encapsulate those changes in a separate PR. 
- I've added a spinner to the generate and disable cells that appears when the respective API calls are being made. 
- I've been a little sneaky and co-opted the availableRoles array to ensure that A) invite links are sorted to match and B) the invite links "Role" selection matches the "Role" selection in section 2. This should be okay as the actual roles should not differ.

To test:

### Non-P2
- [ ] View the invite person screen on a non-p2 site.  Confirm that the invite links section is not shown and no invite links tracks events are dispatched.

### P2
- [ ] View a P2 and confirm you see an option to Generate links.
- [ ] Use the generate feature and confirm that the invite links section updates with options to share and manage links. 
- [ ] Confirm you can change the selected invite link via the Role cell. 
- [ ] Confirm you can share the selected invite link via the share cell. Share options will vary depending on simulator / device.
- [ ] Confirm you can disable links via the disable feature. 
- [ ] Confirm the appropriate tracks events are dispatched. 
- [ ] Confirm the UI matches the design except as noted above.
- [ ] Confirm you see spinners during api requests as expected. 
- [ ] Confirm that row taps are ignored while the feature is busy.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Examples:
![Simulator Screen Shot - iPhone 11 Pro - 2021-03-03 at 16 11 49](https://user-images.githubusercontent.com/1435271/109879730-9afb3080-7c3b-11eb-9575-e9c7b9653075.png)
![Simulator Screen Shot - iPhone 11 Pro - 2021-03-03 at 16 11 51](https://user-images.githubusercontent.com/1435271/109879734-9b93c700-7c3b-11eb-8fe8-b46f6aeab314.png)
![Simulator Screen Shot - iPhone 11 Pro - 2021-03-03 at 16 11 54](https://user-images.githubusercontent.com/1435271/109879738-9c2c5d80-7c3b-11eb-882d-b15af752311d.png)
![Simulator Screen Shot - iPhone 11 Pro - 2021-03-03 at 16 11 57](https://user-images.githubusercontent.com/1435271/109879741-9cc4f400-7c3b-11eb-8d5d-f559d2923869.png)

aerych committed:
Invite Links: Stub a new tableview section in the storyboard.

aerych committed:
Invite Links: Adds section title and footer.

aerych committed:
Merge branch 'develop' of github.com:wordpress-mobile/WordPress-iOS into feature/15356-invite-link

aerych committed:
InviteLinks: Show sections and rows as needed.

aerych committed:
InviteLinks: Update workaround to update only when needed.

aerych committed:
InviteLinks: Stub row handling

aerych committed:
InviteLinks: Wire up basic fetch, generate, disable api calls.

aerych committed:
InviteLinks: Stub refresh methods for other invite cells.

aerych committed:
InviteLinks: Sort links. Refresh selected and expires cells.

aerych committed:
InviteLinks: Wire up invite link role selection.

aerych committed:
InviteLinks: Wire up sharing.

aerych committed:
InviteLinks: Show alert confirming disable links.

aerych committed:
InviteLinks: Add check for expiry value.

aerych committed:
InviteLinks: Show a notice in the case of an error.

aerych committed:
InviteLinks: Sync after generating so expiry is populated.

aerych committed:
InviteLinks: Outlet and setup for disable cell.

aerych committed:
InviteLinks: Only do cell setup once.

aerych committed:
InviteLinks: Show spinner and ignore actions when busy.

aerych committed:
InviteLinks: Wires up analytics.

aerych committed:
InviteLinks: Fetch links after generating.

aerych committed:
InviteLinks: Don't call sync after generating.

aerych commented:
cc @mattmiklic for a design check :)

ScoutHarris commented on WordPress/Classes/ViewRelated/People/InvitePersonViewController.swift:
```diff
@@ -410,6 +560,197 @@ private extension InvitePersonViewController {
+            } else {
+                shareInviteLink()
+            }
+        case .role:
+            // Handled by segue
+            break
+        case .expires:
+            // No op.
+            break
```
Couldn't `default` cover these?

aerych replied:
Why yes. Yes it could. 👍

ScoutHarris reviewed (changes requested):
Hey Eric.

It all looks good except for one thing. The Message footer is now blue, and is a link to the roles definition. 

<img width="490" alt="message_footer" src="https://user-images.githubusercontent.com/1816888/109885227-62f3ed80-7c3b-11eb-881d-4fc1fac66f05.png">

mattmiklic reviewed (changes requested):
@aerych This looks good for the most part, but the "Generate new link" and "Share invite link" buttons are a shade of blue I don't recognize. Could you double-check that they're using our `primary` blue (blue 50 from Color Studio)?

aerych committed:
InviteLinks: Ensure the correct colors are used.

aerych committed:
InviteLinks: Use a default instead of no op case statements.

aerych committed:
InviteLinks: Ensure message footer is not interactive.

aerych commented:
Thanks @ScoutHarris and @mattmiklic!  
I've updated the colors to be sure they are the correct ones. Please confirm :) 
The message section footer should no longer look or behave like a link.
I've cleaned up the switch statement. 

Ready for another peek!

![Simulator Screen Shot - iPhone 11 Pro - 2021-03-04 at 12 28 19](https://user-images.githubusercontent.com/1435271/110011593-3e545000-7ce5-11eb-8edb-f35792832b66.png)

mattmiklic reviewed (approved):
Looks good to me now!

ScoutHarris reviewed (approved):
👍 !

:shipit:

aerych commented:
Thank you all!!!
